modules/global/src/com/haulmont/cuba/core/sys/AbstractViewRepository.java

```diff
@@ -120,14 +125,28 @@ protected void checkDuplicates(Element rootElem) {
             String key = getMetaClass(viewElem) + "/" + viewName;
             if (!Boolean.parseBoolean(viewElem.attributeValue("overwrite"))) {
                 String extend = viewElem.attributeValue("extends");
-                if (!StringUtils.equals(extend, viewName) && checked.contains(key)) {
-                    log.warn("Duplicate view definition without 'overwrite' attribute and not extending parent view: " + key);
+                if (extend != null) {
+                    List<String> ancestors = splitExtends(extend);
+
+                    if (isAncestorsNotConstrainViewName(ancestors, viewName) && checked.contains(key)) {
+                        log.warn("Duplicate view definition without 'overwrite' attribute and not extending parent view: " + key);
+                    }
                 }
             }
             checked.add(key);
         }
     }
 
+    protected List<String> splitExtends(String extend) {
+        return Splitter.on(',').omitEmptyStrings().trimResults().splitToList(extend);
+    }
+
+    protected boolean isAncestorsNotConstrainViewName(List<String> ancestors, String viewName) {

```

Comment:
First, the name is wrong. "Contain", not "constrain". Better name would be "listDoesNotContainViewName".
Second, it can be replaced with `ancestors.contains(viewName)`. So I would suggest to inline and remove the method at all.

Reply:
done